0000-classical-stores.md

```diff
@@ -0,0 +1,516 @@
+
+However, in the end, I have many reasons that I think this would not the correct decision:
+
+- There's 5 pointers of storage space required per Python-space `Clbit` that's just wrapped by a `Var`, not to mention the additional costs of tracking the bit within `QuantumCircuit`.
+  For a `double`, that's over 2kB of extra tracking, even if we do nothing with it.
+- It implies that we need to track each of these `Clbit` wires separately in the DAG, even though the data-flow is bundled - again, lots of unnecessary memory use.
+- There is a natural split in hardware between a `Clbit`, which is a target for a `measure` instruction and is likely written to in controllers very close to the QPU hardware, and classical variables that can be in outer control systems.
+  For example, gathering the result of a `measure` into the particular bit of a `uint[8]` might involve more expensive transfer and synchronisation operations than adding together two `uint[8]` objects.
+- If there are `Clbit` instances backing `Var`, we presumably would need to duplicate this idea of data-tracking into the `.clbits` fields of `CircuitInstruction` and `DAGOpNode`, which means we need to define an "ordering" to how the bits are encountered through every instruction, and because they would automatically be in the `Expr` tree, doing so introduces the possibility of failed data coherence, where the `.clbits` fields do not match the nodes.
+- If we have `Clbit` instances, we cannot use the same methods of variable-closure in nested scopes, and compound instructions taking input parameters or using input parameters only internally would have confusing behaviour between the `.clbits` fields of their own `CircuitInstruction` and their internal `Var` nodes.
+- `ClassicalRegister` would _still_ be special cased, however, because it does not fundamentally "own" its bits; it's possible to have aliased bits in more than one register.
+  This is very fiddly to work with in general, is highly unlikely to have any real hardware support in the near term, and it's unclear how it's useful.
+
+#### Give `Store` first-class support as an alternative to `CircuitInstruction`
+
+The RFC currently proposes to jam `Store` into being a subclass of `Instruction` that has inherent classical bits (similar to `Instruction.condition` or the `ControlFlowOp` subclasses).
+This means that all consumers of `QuantumCircuit` that care about data-flow orderings and display must know to check for this particular subclass, and include custom code to handle this.
+Similarly, the `DAGCircuit` would need to grow an alternative to `DAGOpNode` that could represent this with new first-class support.
+
+A strong alternative would be to have different versions of the `CircuitInstruction`/`DAGOpNode` circuit context object that better represent the "special" classical operations such as `Measure`, `Store` and the `ControlFlowOp` subclasses.
```

Comment:
Worth noting that this would imply that `CircuitData` introduced in https://github.com/Qiskit/qiskit/pull/10827 would need to gain awareness of these special classical operations (i.e. via separate methods like `append_store` or with subclass instance checking through its `append`, in addition to internal tracking in either case).

It's certainly doable, but it's useful to know that `CircuitData` (though currently private) will likely need more awareness/introspection of the instructions it stores, esp. as we oxidize more parts of `QuantumCircuit`.